src/frontends/tensorflow/src/saved_model.hpp

```diff
@@ -0,0 +1,240 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <map>
+
+#include "graph_iterator_proto.hpp"
+#include "openvino/util/file_util.hpp"
+#include "saved_model.pb.h"
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+
+struct VIBlock;
+
+template <typename T>
+std::basic_string<T> get_saved_model_name() {}
+template <typename T>
+std::basic_string<T> get_variables_index_name() {}
+
+template <>
+std::basic_string<char> get_saved_model_name<char>();
+template <>
+std::basic_string<char> get_variables_index_name<char>();
+
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+template <>
+std::basic_string<wchar_t> get_saved_model_name<wchar_t>();
+template <>
+std::basic_string<wchar_t> get_variables_index_name<wchar_t>();
+#endif
+
+// Stores information about variables index
+class SavedModelVariablesIndex {
+    // Contains maximum amount of shards, used for creating corrext extension
+    int32_t m_total_shards;
+    // Contains BundleEntryProto variables list, readed from .index file
+    std::map<std::string, std::vector<char>> m_variables_index;
+    // List of opened data files for using with BundleEntryProto
+    std::map<int32_t, std::shared_ptr<std::ifstream>> m_data_files;
+    // List of mapped variables which could be read using TrackableObjectGraph
+    std::map<std::string, std::string> m_variables_map;
+
+public:
+    /// \brief Reads variables from opened variable index file. Can cause an asserts in case of issues.
+    /// \param vi_stream Opened stream file, file pointer doesn't matter, it will be rewind internally.
+    /// \param path A path to file with variables data
+    /// \returns Returns true in case of everything loads successfully, false otherwise
+    bool read_variables(std::ifstream& vi_stream, const std::string& path);
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+    /// \brief Reads variables from opened variable index file. Can cause an asserts in case of issues.
+    /// \param vi_stream Opened stream file, file pointer doesn't matter, it will be rewind internally.
+    /// \param path A path to file with variables data
+    /// \returns Returns true in case of everything loads successfully, false otherwise
+    bool read_variables(std::ifstream& vi_stream, const std::wstring& path);
+#endif
+
+    /// \brief Returns data and size of data of stored variable
+    /// \param name Name of variable
+    /// \param data Pointer on a pointer where data pointer will be returned
+    /// \param size Pointer on a variable which will stores data size
+    /// \returns Returns true in case variable was found, false otherwise (data and size will be untouched)
+    bool get_variable(const std::string& name, char** data, size_t* size) {
```

Comment:
can we use `std::vector<char>` or `std::vector<uint8_t>` instead of `char** data, size_t* size`?
I think it will be more safe.

Reply:
Read comment below, it is applicable for both methods.

Reply:
Not careful memory management. Propose to use smart pointers.